src/HOI4World/States/HoI4States.cpp

```diff
@@ -606,6 +620,15 @@ void HoI4::States::addProvincesAndCoresToNewState(State& newState,
 		{
 			newState.addClaims({HoI4Core});
 		}
+		Log(LogLevel::Debug) << "State: " << newState.getID();
+		for (const auto& popCulture: popCultureMap)
+		{
+			Log(LogLevel::Debug) << "\t\t\t" << popCulture.first << " Pops: " << popCulture.second
+										<< "\n\tPopulation: " << static_cast<double>(totalPopulation);
+			if (popCulture.second / static_cast<double>(totalPopulation) >= 0.25 &&
+				 !newState.getMajorCultures().count(popCulture.first))
+				newState.addMajorCulture(popCulture.first);
+		}
```

Comment:
Make this conditional on debug being set to true in configuration. You can pass just the debug variable to this function instead of the whole config when doing so.